icepyx/tests/test_behind_NSIDC_API_login.py

```diff
@@ -2,41 +2,31 @@
 import pytest
 import warnings
 
-# from unittest.mock import patch
-# import mock
-# import builtins
-# import getpass
-import os
 
+# Misc notes and needed tests
 # test avail data and subsetting success for each input type (kml, shp, list of coords, bbox)
-# check that agent key is added in event of no subsetting
-# check that downloaded data is subset
+# check that downloaded data is subset? or is this an NSIDC level test so long as we verify the right info is submitted?
 
 
 @pytest.fixture
-def reg_a(scope="module"):
+def reg(scope="module"):
     return ipx.Query("ATL06", [-55, 68, -48, 71], ["2019-02-22", "2019-02-28"])
 
 
-# @patch('my_module.__get_input', return_value='y')
-
-
+# QUESTION: should we be testing independently to make sure the session starts, or is the fixture sufficient test of that?
+# test_earthdata_session_started(an early attempt at the above) doesn't work because the 'requests.sessions.Session' isn't recognized...
+# def test_earthdata_session_started(session):
+#     assert isinstance(session, 'requests.sessions.Session')
 @pytest.fixture
-def session(reg_a, scope="module"):
-    return reg_a._start_earthdata_session(
+def session(reg, scope="module"):
+    return reg.Earthdata._start_session(
         "icepyx_devteam", "icepyx.dev@gmail.com", os.getenv("NSIDC_LOGIN")
     )
 
 
-# QUESTION: should we be testing to make sure the session starts? If so, how? The below doesn't work because the 'requests.sessions.Session' isn't recognized... is this a case where I'd need to have a mock session to compare it to?
-# def test_earthdata_session_started(session):
-#     assert isinstance(session, 'requests.sessions.Session')
-
-
+########## is2ref module ##########
 import icepyx.core.is2ref as is2ref
 
-########## _get_custom_options ##########
-# TestQuestion: should this use a mock rather than an actual call to NSIDC?
 # TestQuestion: is there a better way to deal with this than having the super long dictionary copied here?
```

Comment:
@weiji14

Reply:
If you're referring to the question; "should this use a mock rather than an actual call to NSIDC?", it depends. My preference is to lean towards a simpler solution (don't use a mock) since it's not easy finding people who understand how to set up pytest mocks (maintenance burden). But if the intention (as with https://github.com/icesat2py/icepyx/pull/200#discussion_r642704522) is to **not** ping the CMR system too often, then a mock would be preferable.

If you're referring to the "is there a better way to deal with this than having the super long dictionary copied here?" question. Then I suppose we could store the dictionary in a plain-text JSON or YAML file and load it during the test to compare?